sktime/utils/_testing/estimator_checks.py

```diff
@@ -426,3 +429,31 @@ def check_persistence_via_pickle(Estimator):
     for method in results:
         unpickled_result = getattr(unpickled_estimator, method)(*args[method])
         _assert_almost_equal(results[method], unpickled_result)
+
+
+def check_multiprocessing_determinism(Estimator):
+    if 'n_jobs' in signature(Estimator.__init__).parameters:
+        estimator = _construct_instance(Estimator)
+        fit_args = _make_args(estimator, "fit")
+        ops = ["predict", "predict_proba", "transform"]
```

Comment:
One last comment: We have a NON_STATE_CHANGING_METHODS list at the top of the file, so we use that here:

```python
for method in NON_STATE_CHANGING_METHOD:
    ...
```

Sorry, I thought I had posted this comment before but can't find it anymore. Everything else looks good to go!